src/net_processing.cpp

```diff
@@ -325,6 +326,7 @@ class PeerManagerImpl final : public PeerManager
     ChainstateManager& m_chainman;
     CTxMemPool& m_mempool;
     TxRequestTracker m_txrequest GUARDED_BY(::cs_main);
+    TxRebroadcastHandler m_txrebroadcast;
```

Comment:
Perhaps make this a unique_ptr, and then pass in a boolean to the PeerManagerImpl ctor to indicate whether you want to enable tx rebroadcast and initialize the pointer.

Then, in `UpdatedBlockTip`:

```diff
diff --git a/src/net_processing.cpp b/src/net_processing.cpp
index 7811e3ca40..8bb5e8f87f 100644
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -1392,8 +1392,8 @@ void PeerManagerImpl::UpdatedBlockTip(const CBlockIndex *pindexNew, const CBlock
     }
 
     // Rebroadcast selected mempool transactions
-    if (gArgs.GetArg("-rebroadcast", DEFAULT_REBROADCAST_ENABLED)) {
-        std::vector<TxIds> rebroadcast_txs = m_txrebroadcast.GetRebroadcastTransactions();
+    if (m_txrebroadcast) {
+        std::vector<TxIds> rebroadcast_txs = m_txrebroadcast->GetRebroadcastTransactions();
 
         LOCK(cs_main);
```

parametrizing whether the TxRebroadcastHandler is initialized avoids having to read the global args every block, and should make it the code path more obviously isolated and testable.

Reply:
great idea! done